### PR TITLE
feat: Implement maze builder interface with CSV/JSON import

### PR DESCRIPTION
**Pull request by sarahsatchell**
[](https://www.figma.com/site/0Aj3fjiFkFOtTQ291uUPCP/AI-Self-Organizing-Agents?node-id=0-1&t=OXx9E9x2mFDSVhg6-1)

### CODE REVIEW

**Review by Kushp2022 — APPROVED** [review]

**Comment by Kushp2022** [comment]
Note for future PR: Please List in the title if your PR is  (e.g.. a new Feature (feat) or a bug fix (fix)...)

File: front-end/src/App.tsx [hunk 1]
```diff
@@ -1,44 +1,313 @@
...
+      <Button onClick={() => { handleClear(); setShowBuilder(true); }}>Start Maze</Button>
     </div>
-  )
+  );
+
+  // Comprehensive handler to generate and validate maze from input data
+  const handleGenerateMaze = () => {
+    setError('');
+    let input = inputType === 'csv' ? csv.trim() : json.trim();
+    if (input === '') {
```
Comment (lelili23) [inline front-end/src/App.tsx]: We could consider adding a maze size check here to prevent browser crashes. Currently, the site crashes if the maze input is 1000 x 1000. For ex:
````
const MAX_INPUT_SIZE = 1000000; // 1MB of text
if (input.length > MAX_INPUT_SIZE) {
   setError('Input too large. Maximum 1MB of maze data allowed.');
   return;
}
````

File: front-end/src/App.tsx [hunk 2]
```diff
@@ -1,44 +1,313 @@
...
+      return;
+    }
+    if (endPtTemp[0] < 0 || endPtTemp[0] >= rows || endPtTemp[1] < 0 || endPtTemp[1] >= cols) {
+      setError('End point is out of bounds.');
+      return;
+    }
+    if (startPtTemp[0] === endPtTemp[0] && startPtTemp[1] === endPtTemp[1]) {
+      setError('Start and end points cannot be the same.');
+      return;
+    }
```
Comment (lelili23) [inline front-end/src/App.tsx]: We should also check if the start/end points are on a wall, since this shouldn't be possible. For ex: 
```
if (m[startPtTemp[0]][startPtTemp[1]] === 1 || m[endPtTemp[0]][endPtTemp[1]] === 1) {
    setError('Start and end points can't be on walls (1).');
    return;
}
```

File: front-end/src/App.tsx [hunk 3]
```diff
@@ -1,44 +1,313 @@
...
+      } else {
+        setJson(text);
+      }
+    };
+    reader.readAsText(file);
+  };
 
+  if (showBuilder) {
+    // Render the Maze Builder interface
+    return (
```
Comment (lelili23) [inline front-end/src/App.tsx]: The App.tsx is starting to get long and handle multiple things, we could consider refactoring the maze builder section into its own component to keep things organized

File: front-end/src/App.tsx [hunk 4]
```diff
@@ -1,44 +1,313 @@
-import oregonLogo from './assets/Oregon_State_text_logo.png'
-import './App.css'
-import Header from './components/Header'
-import Button from './components/Button'
-import mazeImg from './assets/maze.png'
+
```
Comment (lelili23) [inline front-end/src/App.tsx]: Now that the maze input is being sent to the backend in the recent PRs, we should add unit tests for the maze submission. Could cover parsing, validation, and edge cases tests. We could add a new ticket for this task

File: front-end/src/App.tsx (hunk 3 above)
Comment (sarahsatchell) [reply front-end/src/App.tsx]: Sounds good, will create future PR including this. Thanks!
